experimental/SetPartitions/src/PartitionProperties.jl

```diff
@@ -0,0 +1,194 @@
+function size(p::SetPartition)
+    return length(p.lower_points) + length(p.upper_points)
+end
+
+function is_composable(p::SetPartition, q::SetPartition)
+    return length(p.upper_points) == length(q.lower_points)
+end
+
+function upper_points(p::SetPartition)
+    return p.upper_points
+end
+
+function lower_points(p::SetPartition)
+    return p.lower_points
+end
+
+function size(p::ColoredPartition)
+    return size(p.partition)
+end
+
+function is_composable(p::ColoredPartition, q::ColoredPartition)
+    return p.color_upper_points == q.color_lower_points && 
+        is_composable(p.partition, q.partition)
+end
+
+function upper_points(p::ColoredPartition)
+    return p.partition.upper_points
```

Comment:
```suggestion
    return upper_points(p.partition)
```
avoid direct access as much as possible

Reply:
not just here, but basically everywhere. In the optimal case, you only have direct access in one function per field and everything else refers to that.